zenodo-tests/test_prep/test_stripe.py

```diff
@@ -0,0 +1,176 @@
+            92.4986,
+        ),
+        (
+            "i12_dataset4",
+            31,
+            99.3062,
+        ),
+    ],
+    ids=["size_11", "size_21", "size_31"],
+)
+def test_remove_stripe_based_sorting_i12_dataset4(
+    request, dataset_fixture, size_filt, norm_res_expected
+):
+    dataset = request.getfixturevalue(dataset_fixture)
+    data_normalised = normalize(dataset[0], dataset[2], dataset[3], minus_log=True)
+
+    del dataset
+    force_clean_gpu_memory()
+
+    output = remove_stripe_based_sorting(
+        cp.copy(data_normalised), size=size_filt, dim=1
+    )
+
+    residual_calc = data_normalised - output
+    norm_res = np.linalg.norm(residual_calc.get().flatten())
+
+    assert isclose(norm_res, norm_res_expected, abs_tol=10**-4)
+
+    assert output.dtype == np.float32
+    assert output.flags.c_contiguous
+
+
+@pytest.mark.parametrize(
+    "dataset_fixture, beta_val, norm_res_expected",
+    [
+        (
+            "i12_dataset4",
+            0.01,
+            321.5298,
+        ),
+        (
+            "i12_dataset4",
+            0.03,
+            172.1631,
+        ),
+        (
+            "i12_dataset4",
+            0.06,
+            128.0331,
+        ),
+    ],
+    ids=["beta_001", "beta_003", "beta_006"],
+)
+def test_remove_stripe_ti_i12_dataset4(
+    request, dataset_fixture, beta_val, norm_res_expected
+):
+    dataset = request.getfixturevalue(dataset_fixture)
+    data_normalised = normalize(dataset[0], dataset[2], dataset[3], minus_log=True)
+
+    del dataset
+    force_clean_gpu_memory()
+
+    output = remove_stripe_ti(cp.copy(data_normalised), beta=beta_val)
+
+    residual_calc = data_normalised - output
+    norm_res = np.linalg.norm(residual_calc.get().flatten())
+
+    assert isclose(norm_res, norm_res_expected, abs_tol=10**-4)
+
+    assert output.dtype == np.float32
+    assert output.flags.c_contiguous
+
+
+@pytest.mark.parametrize(
+    "dataset_fixture, snr_val, la_size_val, sm_size_val, norm_res_expected",
+    [
+        ("i12_dataset4", 1.0, 31, 10, 104.8582),
+        ("i12_dataset4", 2.0, 41, 17, 99.4554),
+        ("i12_dataset4", 3.0, 61, 21, 102.8688),
+        ("i12_dataset4", 4.0, 71, 31, 106.1418),
+    ],
+    ids=["snr_1", "snr_2", "snr_3", "snr_4"],
+)
+def test_remove_all_stripe_i12_dataset4(
+    request, dataset_fixture, snr_val, la_size_val, sm_size_val, norm_res_expected
+):
+    dataset = request.getfixturevalue(dataset_fixture)
+    data_normalised = normalize(dataset[0], dataset[2], dataset[3], minus_log=True)
+
+    del dataset
+    force_clean_gpu_memory()
+
+    output = remove_all_stripe(
+        cp.copy(data_normalised),
+        snr=snr_val,
+        la_size=la_size_val,
+        sm_size=sm_size_val,
+        dim=1,
+    )
+
+    residual_calc = data_normalised - output
+    norm_res = np.linalg.norm(residual_calc.get().flatten())
+
+    assert isclose(norm_res, norm_res_expected, abs_tol=10**-4)
+
+    assert output.dtype == np.float32
```

Comment:
All looks good. Just an extra sanity check!
```suggestion
    assert isclose(norm_res, norm_res_expected, abs_tol=10**-4)
    assert not np.isnan(output).any(), "Output contains NaN values"
    assert output.dtype == np.float32
```